packages/expo-media-library/android/src/main/java/expo/modules/medialibrary/MediaLibraryModule.kt

```diff
@@ -457,6 +457,31 @@ class MediaLibraryModule : Module() {
     action.runWithPermissions(true)
   }
 
+  private fun runActionWithDeletePermissions(assetsId: List<String>, action: Action) {
```

Comment:
In `runActionWithPermissions` we check for paths without permisison. I noticed that in your implementation we assume we don't have the permissions. Is this because Android will always require the permission or is there another reason?

Also `runActionWithDeletePermissions` and `runActionWithPermissions` are really similar, maybe it would be possible to add a parameter to `runActionWithPermissions` such as `useDeletePermissions` in order to avoid using a separate function?

Reply:
Thanks for the review. 
1.) You are right. That can be a scenario. I was just confused with `Intent.FLAG_GRANT_WRITE_URI_PERMISSION`. I will put permission check back.
2.) I will try combine it with `runActionWithPermissions `.